### PR TITLE
Update to beta4

### PR DESCRIPTION
**Pull request by timbussmann**
probably easier to review the individual commits because of the ATT noise.

note that I had to add several saga configs due to failing tests. I added invalid mappings to just satisfy the checks and get the tests green.

@Particular/nhibernate-persistence-maintainers please review

### CODE REVIEW

File: src/NServiceBus.NHibernate.AcceptanceTests-Oracle/App_Packages/NSB.AcceptanceTests.6.0.0-beta0004/Sagas/When_a_finder_exists_and_found_saga.cs [hunk 1]
```diff
@@ -71,7 +71,6 @@ public Task Handle(SomeOtherMessage message, IMessageHandlerContext context)
                 protected override void ConfigureHowToFindSaga(SagaPropertyMapper<SagaData08> mapper)
                 {
                     mapper.ConfigureMapping<StartSagaMessage>(saga => saga.Property).ToSaga(saga => saga.Property);
-                    // Mapping not required for SomeOtherMessage because CustomFinder used
```
Comment (DavidBoike) [inline src/NServiceBus.NHibernate.AcceptanceTests-Oracle/App_Packages/NSB.AcceptanceTests.6.0.0-beta0004/Sagas/When_a_finder_exists_and_found_saga.cs]: Why remove this?

Comment (timbussmann) [inline src/NServiceBus.NHibernate.AcceptanceTests-Oracle/App_Packages/NSB.AcceptanceTests.6.0.0-beta0004/Sagas/When_a_finder_exists_and_found_saga.cs]: good catch. Probably removed to much. Can readd that.

**Comment by DavidBoike** [comment]
it looks a little odd mapping IMessage's GetHashCode() (an int) to the Saga's Id (a Guid) but the point of those tests is just to determine how NHibernate will map them. I think instantiating SagaMetadata on those would fail (different types) but I can't see adding a bunch of cruft for those things which should really be tested by the core anyway.

This LGTM.

@Particular/nhibernate-persistence-maintainers will you please provide the official merge?

**Comment by HEskandari** [comment]
This LGTMT! Merging...

**Pull request merged**